### DISCUSSION TITLE
How to avoid 2 thirdparty lib all using stb_image caused duplicated symbol collision?

### DISCUSSION THREAD

**Discussion by lucasjinreal** [nothings/stb#1430] [Q&A]
How to avoid 2 thirdparty lib all using stb_image caused duplicated symbol collision?

**Comment by nothings** [comment]
This is the fault of the third party libraries, and you should report to them to use STB_IMAGE_STATIC to avoid this problem.

If it's only a single library, as a workaround, YOU can use STB_IMAGE_STATIC: define it in the same file as STB_IMAGE_IMPLEMENTATION. the functions will then only be visible in that file. If you need them to be visible in other files, write wrapper functions with new names that call the static functions.

If the problem is that two different third-party libraries are conflicting, then there is nothing that can be done without changing at least one of those libraries.
